sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -565,8 +569,17 @@ case class DescribeTableCommand(
     result
   }
 
+  private def schemaWithoutPartCols(schema: StructType, partColNames: Seq[String]): StructType = {
+    if (partColNames.isEmpty) {
+      schema
+    } else {
+      StructType(schema.filter(col => !partColNames.contains(col.name)))
```

Comment:
Although this function is used for the data from `metadata` only for now, this is a usual suspect for the case-sensitive failures.

We don't know who will use this function later. And, since `partColNames` is `Seq[String]`, it's not safe. We had better consider a case-sensitivity. Please refer `SchemaUtils.normalizePartitionSpec` and `SchemaUtils.checkColumnNameDuplication`. You can use `resolver` to be robust.

Reply:
@dongjoon-hyun  
Updated,  current way is robust enough. Since here is just show table desc. use resolver to check column name equal.
By the way :  `PartitionUtils.normalizePartitionSpec`